### PR TITLE
Aggregator Recompute cert hash command

### LINKED ISSUE #1001: Add `recompute-certificates-hash` command to aggregator
## Issue
In order to simplify the handling of changes in the certificates (as in #946) which currently require a regenesis of the certificate chain, we need to have a command line tool to recompute the hashes of the certificates once they are migrated.

## To do

- [x] Add a `recompute-certificates-hash` in a `tools` command that recomputes the hash of the certificates
- [x] Update documentation https://mithril.network/doc/manual/developer-docs/nodes/mithril-aggregator/?

## Related issue
#946

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR add a new commands that make the aggregator recompute all its certificates hash (and update certificate related entities such as `signed_entity`).
This allow us to change the structure of the `Certificate` and `CertificateMetadata` types without the needs to re-genesis the certificate chain.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Comments
This PR also split the aggregator commands arguments in a dedicated sub-module/

## Issue(s)
Closes #1001

### CODE REVIEW

File: mithril-aggregator/src/tools/certificates_hash_migrator.rs [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+            certificate.hash = new_hash;
+            migrated_certificates.push(certificate);
+        }
+
+        // 2 - Update all certificates previous hash
+        for mut certificate in migrated_certificates.iter_mut() {
+            // Should we raise an error if no matching is found ?
+            if let Some(new_hash) = old_and_new_hashes.get(&certificate.previous_hash) {
+                certificate.previous_hash = new_hash.clone();
```
Comment (jpraynaud) [inline mithril-aggregator/src/tools/certificates_hash_migrator.rs]: After this line, the certificate hash should be updated as well

File: docs/root/manual/developer-docs/nodes/mithril-aggregator.md [hunk 2]
```diff
@@ -436,4 +474,6 @@ General parameters:
 |-----------|---------------------|:---------------------:|----------------------|-------------|---------------|---------|:---------:|
 | `current_era_epoch` | `--current-era-epoch` | - | `CURRENT_ERA_EPOCH` | Epoch at which current era starts. | - | - | - | :heavy_check_mark: |
 | `next_era_epoch` | `--next-era-epoch` | - | `NEXT_ERA_EPOCH` | Epoch at which the next era starts. If not specified and an upcoming era is available, it will announce the next era. If specified, it must be strictly greater than `current-epoch-era` | - | - | - | - |
-| `era_markers_secret_key` | `--era-markers-secret-key` | - | `ERA_MARKERS_SECRET_KEY` | Era Markers Secret Key that is used to verify the authenticity of the era markers on chain. | - | - | - | :heavy_check_mark: |
\ No newline at end of file
+| `era_markers_secret_key` | `--era-markers-secret-key` | - | `ERA_MARKERS_SECRET_KEY` | Era Markers Secret Key that is used to verify the authenticity of the era markers on chain. | - | - | - | :heavy_check_mark: |
+
+`tools recompute-certificates-hash` command have no dedicated parameter
```
Comment (jpraynaud) [inline docs/root/manual/developer-docs/nodes/mithril-aggregator.md:479]: Typo here: `has no dedicated parameter`

File: mithril-aggregator/src/tools/certificates_hash_migrator.rs [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+        let old_certificates = self
+            .certificate_repository
+            // arbitrary high value to get all existing certificates
+            .get_latest_certificates(usize::MAX)
+            .await?;
+
+        let mut migrated_certificates = vec![];
+        let mut old_and_new_hashes = HashMap::new();
+
+        // 1 - Recompute all certificates hashes, but not their previous
```
[the remark this answers is not in the document]
Comment (Alenar) [reply mithril-aggregator/src/tools/certificates_hash_migrator.rs:54]: Yup, I completely overlook that, a patch is coming soon to fix this.

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Review by ghubertpalo — APPROVED** [review]

**Pull request merged**